database/src/lib.rs

```diff
@@ -23,3 +23,9 @@ pub trait Database {
         prefix: Vec<u8>,
     ) -> Box<dyn Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a>;
 }
+
+pub trait DatabaseBuilder<DB> {
```

Comment:
Would it make more sense for the DB to be an associated type?

Reply:
I thought about it, but think it's easier to use simple generic + generic allows different implementations.
I mean that you could use single structure for different databases.

Reply:
Fair enough.